pkg/controller/status/status_controller.go

```diff
@@ -0,0 +1,152 @@
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+// Reconcile handles the creation and update of configMaps as well as the initial daemonSets for the AIDE pods.
+func (r *ReconcileFileIntegrityStatus) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("reconciling FileIntegrityStatus")
+
+	// Fetch the FileIntegrity instance
+	instance := &fileintegrityv1alpha1.FileIntegrity{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if kerr.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// first check is for the reinit daemonset, when this exists at all we are in an initialization phase.
+	reinitDS := &appsv1.DaemonSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: common.ReinitDaemonSetName, Namespace: common.FileIntegrityNamespace}, reinitDS)
+	if err != nil && !kerr.IsNotFound(err) {
+		reqLogger.Error(err, "error getting reinit daemonSet")
+		return reconcile.Result{}, err
+	}
+	if err == nil {
+		// reinit daemonset is active, thus we are initializing
+		err := updateStatus(r.client, instance, fileintegrityv1alpha1.PhaseInitializing)
+		if err != nil {
+			reqLogger.Error(err, "error updating FileIntegrity status")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: statusRequeue}, nil
+	}
+
+	ds := &appsv1.DaemonSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: common.DaemonSetName, Namespace: common.FileIntegrityNamespace}, ds)
+	if err != nil && !kerr.IsNotFound(err) {
+		reqLogger.Error(err, "error getting daemonSet")
+		return reconcile.Result{}, err
+	}
+
+	if err == nil {
+		if common.DaemonSetIsReady(ds) && !common.DaemonSetIsUpdating(ds) {
+			err := updateStatus(r.client, instance, fileintegrityv1alpha1.PhaseActive)
+			if err != nil {
+				reqLogger.Error(err, "error updating FileIntegrity status")
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{RequeueAfter: statusRequeue}, nil
+		}
+		// Not ready, set to initializing
+		err := updateStatus(r.client, instance, fileintegrityv1alpha1.PhaseInitializing)
+		if err != nil {
+			reqLogger.Error(err, "error updating FileIntegrity status")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: statusRequeue}, nil
+	}
+
+	// both daemonSets were missing, so we're currently inactive.
```

Comment:
In what case would we be inactive? isn't it an issue if the DS's would be missing?